src/mavlink-router/mainloop.cpp

```diff
@@ -742,6 +742,18 @@ bool Mainloop::_retry_timeout_cb(void *data)
 
 void Mainloop::_handle_pipe()
 {
+    enum CMD_ARG_INDEX {
+      CMD = 0,
+      PROTOCOL = 1,
+      NAME = 2,
+      IP = 3,
+      PORT = 4,
+      EAVESDROPPING = 5,
+      COALESCE_BYTES = 6,
+      COALESCE_MS = 7,
+      COALESCE_NODELAY = 8,
+    };
+
     char cmd[1024];
     ssize_t num_read = read(_pipefd, cmd, sizeof(cmd) - 1);
```

Comment:
NB there seems implicit assumption below that you are going to read complete commands from pipe, and also some inner handling of splitting lines. Would advise against that structure (not asking to *necessarily* change, but for purposes of making this testable...)

Reply:
I guess it assumes that the whole thing comes through on the pipe together.